homeassistant/components/homekit/type_media_players.py

```diff
@@ -142,3 +171,184 @@ def update_state(self, new_state):
                               self.entity_id, current_state)
                 self.chars[FEATURE_TOGGLE_MUTE].set_value(current_state)
             self._flag[FEATURE_TOGGLE_MUTE] = False
+
+
+@TYPES.register('TelevisionMediaPlayer')
+class TelevisionMediaPlayer(HomeAccessory):
+    """Generate a Television Media Player accessory."""
+
+    def __init__(self, *args):
+        """Initialize a Switch accessory object."""
+        super().__init__(*args, category=CATEGORY_TELEVISION)
+
+        self._flag = {CHAR_ACTIVE: False, CHAR_ACTIVE_IDENTIFIER: False,
+                      FEATURE_TOGGLE_MUTE: False}
+        self.support_select_source = False
+        self.support_volume_level = False
+
+        self.sources = []
+
+        # Add additional characteristics if volume or input selection supported
+        self.chars_tv = []
+        self.chars_speaker = []
+        features = self.hass.states.get(self.entity_id) \
+            .attributes.get(ATTR_SUPPORTED_FEATURES, 0)
+
+        if features & (SUPPORT_PLAY | SUPPORT_PAUSE):
+            self.chars_tv.append(CHAR_REMOTE_KEY)
+        if features & SUPPORT_VOLUME_MUTE or features & SUPPORT_VOLUME_STEP:
+            self.chars_speaker.extend((CHAR_NAME, CHAR_ACTIVE,
+                                       CHAR_VOLUME_CONTROL_TYPE,
+                                       CHAR_VOLUME_SELECTOR))
+            if features & SUPPORT_VOLUME_SET:
+                self.support_volume_level = True
+                self.chars_speaker.append(CHAR_VOLUME)
+
+        if features & SUPPORT_SELECT_SOURCE:
+            self.support_select_source = True
+
+        serv_tv = self.add_preload_service(SERV_TELEVISION, self.chars_tv)
+        self.set_primary_service(serv_tv)
+        serv_tv.configure_char(CHAR_CONFIGURED_NAME, value=self.display_name)
+        serv_tv.configure_char(CHAR_SLEEP_DISCOVER_MODE, value=True)
+        self.char_active = serv_tv.configure_char(
+            CHAR_ACTIVE, setter_callback=self.set_on_off)
+
+        if CHAR_REMOTE_KEY in self.chars_tv:
+            self.char_remote_key = serv_tv.configure_char(
+                CHAR_REMOTE_KEY, setter_callback=self.set_remote_key)
+
+        if CHAR_VOLUME_SELECTOR in self.chars_speaker:
+            serv_speaker = self.add_preload_service(
+                SERV_TELEVISION_SPEAKER, self.chars_speaker)
+            serv_tv.add_linked_service(serv_speaker)
+
+            name = '{} {}'.format(self.display_name, 'Volume')
+            serv_speaker.configure_char(CHAR_NAME, value=name)
+            serv_speaker.configure_char(CHAR_ACTIVE, value=1)
+
+            self.char_mute = serv_speaker.configure_char(
+                CHAR_MUTE, value=False, setter_callback=self.set_mute)
+
+            volume_control_type = 1 if self.support_volume_level else 2
```

Comment:
```suggestion
            volume_control_type = 1 if CHAR_VOLUME in self.chars_speaker else 2
```
You might need to add a line break